File: herejong-app/src/app/model/user-profile.ts
import { CompanyProfile } from './company-profile';

export class UserProfile {
    id :string ;
    userId: string;
    displayName: string;
    pictureUrl: string;
    statusMassage: string;
    email: string;
    point: string;
    companyId : CompanyProfile ;
    companyProfile : CompanyProfile ;
   
    constructor(
        userId: string, 
        displayName: string, 
        pictureUrl: string, 
        statusMassage: string, 
        email: string, 
        
        companyProfile: CompanyProfile, 
      
   ) {

        this.userId = userId;
        this.displayName = displayName;
        this.pictureUrl =pictureUrl;
        this.statusMassage = statusMassage;
        this.email = email;
        this.companyProfile = companyProfile;
       
       
    }
}

File: herejong-back/herejongAdmin/src/main/java/com/herejong/herejongAdmin/model/UserProfile.java
package com.herejong.herejongAdmin.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "userProfile")
public class UserProfile {

    @Id
	private String  id ;
	@Indexed
    private String userId ;
    private String displayName ;
    private String pictureUrl ;
    private String statusMassage ;
    private String email ;
    private Integer point ;
    private String companyId ;
    private CompanyProfile companyProfile ;

    

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(String displayName) {
        this.displayName = displayName;
    }

    public String getPictureUrl() {
        return pictureUrl;
    }

    public void setPictureUrl(String pictureUrl) {
        this.pictureUrl = pictureUrl;
    }

    public String getStatusMassage() {
        return statusMassage;
    }

    public void setStatusMassage(String statusMassage) {
        this.statusMassage = statusMassage;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public CompanyProfile getCompanyProfile() {
        return companyProfile;
    }

    public void setCompanyProfile(CompanyProfile companyProfile) {
        this.companyProfile = companyProfile;
    }

    public String getCompanyId() {
        return companyId;
    }

    public void setCompanyId(String companyId) {
        this.companyId = companyId;
    }

    public Integer getPoint() {
        return point;
    }

    public void setPoint(Integer point) {
        this.point = point;
    }


    

    
    
    
}